spec.html

```diff
@@ -30185,6 +30197,8 @@ <h1>
               1. Let _bodySym_ be the grammar symbol |AsyncGeneratorBody|.
               1. Let _parameterSym_ be the grammar symbol |FormalParameters[+Yield, +Await]|.
               1. Let _fallbackProto_ be *"%AsyncGeneratorFunction.prototype%"*.
+            1. Let _currentRealm_ be the current Realm Record.
+            1. Perform ? HostEnsureCanCompileStrings(_currentRealm_, _bodySym_, _bodyArg_).
```

Comment:
I'm curious why `_bodyArg_` is submitted for approval but not `_parameterArgs_`. A FormalParameter can include an Initializer; doesn't that make it as 'suspect' as a FunctionBody?

Moreover, is it possible that there are security policies that can't be enforced unless you have both the body *and* the parameters? That is, I'm wondering if the HostEnsureCanCompileStrings call should be delayed until you can pass `_sourceString_` to it.

Reply:
Tricky. You are absolutely right about _parameterArgs_ potentially being executable code. This maybe needs some more thinking about what the developer experience should look like in regard to https://github.com/w3c/webappsec-csp/issues/623. It seems pretty bad if the bundling tool has to calculate its own version of _sourceString_ in order to hash that.

If we did move the call later and pass _sourceString_, we'd also still have to pass _bodyArg_, otherwise the host wouldn't be able to distinguish when the code came from a `TrustedScript`.

Reply:
So, one possibility is that we leave out Trusted Types as a use case for the time being, to be accommodated with a future normative PR. (If this PR was going to accommodate Trusted Types correctly, it'd have to remove step 2 of [PerformEval](https://tc39.es/ecma262/#sec-performeval) anyway, I realized.)

Another possibility being discussed on the CSP side is that `unsafe-eval-hashes` doesn't necessarily have to allow the multi-argument `new Function()` at all, so if no other host was going to make a distinction based on the hashes of the parameter args, we could get away with passing a boolean _hasParameterArgs_, and hosts with CSP could just choose to deny any source string for which that argument was true.

Reply:
The former seems best given that trusted types hasn’t been discussed in awhile nor enthusiastically received in plenary.